sympy/codegen/ast.py

```diff
@@ -452,15 +465,18 @@ def body(self):
 class Token(Basic):
     """ Similar to Symbol, but takes no assumptions.
 
-    Defining fields are set in __slots__.
+    Defining fields are set in __slots__. Attributes (defined in __slots__) are
+    allowed to contain instances of Basic or strings.
     """
 
     __slots__ = []
 
     def __new__(cls, *args, **kwargs):
         if len(args) == 1 and not kwargs and isinstance(args[0], cls):
             return args[0]
-        args = args + tuple([kwargs[k] for k in cls.__slots__[len(args):]])
+        args = args + tuple([kwargs.pop(k) for k in cls.__slots__[len(args):]])
+        if kwargs:
+            raise ValueError("Unknown kwargs: %s" % kwargs)
         obj = Basic.__new__(cls)
```

Comment:
I was playing with this and it's a bit odd that the typical ast node is completely atomic from a SymPy point of view, because nothing is stored in .args. I think .args should be used whereever possible to store the ast node expression tree.